### PR TITLE
[Chore] Added Cleanup for Launch Configurations

### PR DESCRIPTION
**Pull request by okankoAMZ**
# Description of the issue
Currently we do not clean-up after launch configurations are used by our integration test. This leads to waste of resources and also failures in integration test due to maximum amount of launch configurations being reached. 

# Description of changes
* Added a new script that checks our launch configurations and if they are more than 2 months old deletes them. 
* Added  workflow so that this script is triggered with the other cleanup scripts. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
I ran `go run ./clean_launch_configuration/clean_launch_configuration.go`(added a break so it only deletes one)  and it deleted the launch configurations. 
Before the test: 
Screenshot 2024-09-04 at 12 42 59 PM
After the test: 
Screenshot 2024-09-04 at 12 43 14 PM

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: .github/workflows/clean-aws-resources.yml [hunk 1]
```diff
@@ -251,4 +269,4 @@ jobs:
 
       - name: Clean old IAM roles
         working-directory: tool/clean
-        run: go run ./clean_iam_roles/clean_iam_roles.go --tags=clean
+        run: go run ./clean_iam_roles/clean_iam_roles.go --tags=cleanfame
```
Comment (jefchien) [inline .github/workflows/clean-aws-resources.yml:272]: Typo.

**Review by jefchien — DISMISSED** [review]

**Comment by jefchien** [comment]
https://github.com/aws/amazon-cloudwatch-agent/actions/runs/10709352868/job/29693796813

**Review by mitali-salvi — APPROVED** [review]

**Review by Paramadon — APPROVED** [review]

**Review by musa-asad — APPROVED** [review]
